### ISSUE TITLE
Search Results cannot be sorted by a date field

### ISSUE THREAD

**Issue by miyachan** [lnx-search/lnx#26]
Attempting to perform a search where the `order_by` field is a date leads to an error:

```json
{"data":"Schema error: 'Field \"ts\" is of type I64!=Date'","status":400}
```

It looks this is because the `FieldValue` is implied to be i64:

https://github.com/lnx-search/lnx/blob/8d38d386de3ecdb1bced4f55032673860feaccf4/engine/src/index/reader.rs#L549-L552

**Comment by miyachan** [comment]
Actual this looks like it might be documentation issue (in tanvity?). According to https://docs.rs/tantivy/0.16.0/tantivy/fastfield/trait.FastValue.html dates cannot be fast fields.

**Comment by ChillFish8** [comment]
Date fields are fast fields iirc because they're stored as a i64 timestamp internally, or at least according to the docs so its interesting that this behavour doesnt work.
This might require some seperate testing to see if dates requires some other special handling or not but the behavour seems weird.

**Comment by ChillFish8** [comment]
I have an idea of what Could be causing this but I'll rewuire some test later on today, if im right the type magic might be failing us here :(

**Comment by ChillFish8** [comment]
Okay, this should be fixed now, this was a consequence of a misleading error due to tantivy not having chrono's serde feature enabled.
The fix has been added to the 0.6.x branch (master) and will be backported to 0.5.1
